src/Illuminate/Collections/Arr.php

```diff
@@ -896,4 +896,49 @@ public static function wrap($value)
 
         return is_array($value) ? $value : [$value];
     }
+
+	/**
+	 * Generate all possible combinations from an array (permutations) array elements included.
+	 *
+	 * @param array $array
+	 * @param bool  $includeOriginalArray
+	 *
+	 * @return array
+	 */
+	public static function permutations(array $array = [], bool $includeOriginalArray = false)
+	{
+
+		$permutations = $includeOriginalArray ? $array : [];
+		$size = count($array);
+
+		$indices = [];
+		for ($index = 0; $index < $size; $index++) {
+			$indices[$index] = 0;
+		}
+
+		while (1) {
+
+			$res = [];
+			for ($index = 0; $index < $size; $index++) {
+				$res[] = $array[$index][$indices[$index]];
+			}
+
+			$permutations[] = $res;
+			$next = $size - 1;
+			while ($next >= 0 &&
+				   ($indices[$next] + 1 >= count($array[$next]))) {
```

Comment:
@elminson Just to make it clearer that the addition should happen first
```suggestion
				   (($indices[$next] + 1) >= count($array[$next]))) {
```

Reply:
make sense to me @morloderex, done!